cli/cmd/set.go

```diff
@@ -231,7 +231,7 @@ func policyMapToBytesList(m []deployment) [][]byte {
 	return policies
 }
 
-func loadWorkloadOwnerKey(path string, manifst *manifest.Manifest, log *slog.Logger) (*ecdsa.PrivateKey, error) {
+func loadWorkloadOwnerKey(path string, manifst *manifest.Manifest, log *slog.Logger) (crypto.PrivateKey, error) {
```

Comment:
I think the underlying error here is returning `nil, nil` and then using the non-error value as if nothing happened.
I think this function should simply return the error of `os.ReadFile()` (Note that we currently also continue on all errors that are not a IsNotExists error). Then in the caller, we check for `os.IsNotExist()` and the use a nil value without type as the privateKey.

Reply:
Oh yes, at least there's an error check missing.
Regarding your suggestion, I think both yours and the current implementation can make sense. (For the latter, nil is a valid workload owner key semantically for the set command.) I don't have a preference. @katexochen?

Reply:
I agree that we should fix the error and work with that. Seems to be the least error prone to do.